### PR TITLE
Feat add regional options

### PR DESCRIPTION
**Pull request by mansueli**
## What kind of change does this PR introduce?

Improve DX for regional Edge Function invocation. 

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: src/FunctionsClient.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ import {
 export class FunctionsClient {
   protected url: string
   protected headers: Record<string, string>
+  protected region: string
```
Comment (laktek) [inline src/FunctionsClient.ts:14]: hmm..what's the API to set the `region`?

File: src/types.ts [hunk 2]
```diff
@@ -50,6 +50,25 @@ export type FunctionInvokeOptions = {
    * The HTTP verb of the request
    */
   method?: 'POST' | 'GET' | 'PUT' | 'PATCH' | 'DELETE'
+  /**
+   * The Region to invoke the function in.
+   */
+  region?:
+    | 'any'
+    | 'ap-northeast-1'
```
Comment (laktek) [inline src/types.ts:58]: let's move these values to an enum.

File: src/FunctionsClient.ts (hunk 1 above)
Comment (mansueli) [reply src/FunctionsClient.ts:14]: I am not sure I follow.

**Comment by laktek** [comment]
@mansueli Can we add a test so the test coverage hits the target? Let me know if you need help with it.

File: test/spec/params.spec.ts [hunk 3]
```diff
@@ -154,21 +154,125 @@ describe('params reached to function', () => {
...
+      ContentType.TEXT
+    )
+    console.log(data?.headers)
+    expect(
+      (data?.headers as [Array<string>]).filter(([k, v]) => k === 'x-region' && v === validRegion)
+        .length > 0
+    ).toBe(true)
+  })
+
+  test('invoke with region overrides region in the client', async () => {
```
Comment (egor-romanov) [inline test/spec/params.spec.ts:233]: @mansueli can you please rename test to reflect the idea?

i just drafted these new ones, so would be great if all new tests will have meaningful names

**Review by laktek — APPROVED** [review]

File: test/spec/params.spec.ts [hunk 4]
```diff
@@ -146,6 +146,158 @@ describe('params reached to function', () => {
...
+      ContentType.TEXT
+    )
+    console.log(data?.headers)
+    expect(
+      (data?.headers as [Array<string>]).filter(([k, v]) => k === 'x-region' && v === validRegion)
+        .length > 0
+    ).toBe(true)
+  })
+
+  test('invoke mirror with invoke header and valid region', async () => {
```
Comment (egor-romanov) [inline test/spec/params.spec.ts:275]: @mansueli also this one duplicates the first one (line 149). 

wdyt about adding a new test that will check that if you set some valid region on the client level (constructor), but then override it on the invoke level to "any", there will be no header in the request?

Comment (mansueli) [reply test/spec/params.spec.ts]: Done:
https://github.com/supabase/functions-js/pull/75/files#diff-8cfc42c763162f3b42260d46eef32a3bb8ba2e0cccadf486353af403526946e6R233

File: test/spec/params.spec.ts (hunk 3 above)
Comment (mansueli) [reply test/spec/params.spec.ts]: I've updated the names of the tests

**Review by egor-romanov — APPROVED** [review]

**Comment by egor-romanov** [comment]
👍

**Pull request merged**